### PR TITLE
Fix the ETLTemplate.configureAdapter to set the right sourcePluginId and sinkPluginId

### PR DESCRIPTION
**Pull request by henry-cask**
This also contains fix for JMS plugin to set the right class loader when using external
dependencies.

### CODE REVIEW

**Comment by henry-cask** [comment]
CC @chtyim and @gokulavasan

File: cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/co/cask/cdap/template/etl/common/ETLTemplate.java [hunk 1]
```diff
@@ -120,8 +120,8 @@ public void configureAdapter(String adapterName, T etlConfig, AdapterConfigurer
     // Validate Source -> Transform -> Sink hookup
     validateStages(source, sink, transforms);
 
-    configure(source, configurer, Constants.Source.PLUGINID);
-    configure(sink, configurer, Constants.Sink.PLUGINID);
+    configure(source, configurer, sourcePluginId);
```
Comment (bdmogal) [inline cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/co/cask/cdap/template/etl/common/ETLTemplate.java]: This is already being done in #2759

Comment (henry-cask) [inline cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/co/cask/cdap/template/etl/common/ETLTemplate.java]: Ah ok will merge the change

**Comment by chtyim** [comment]
LGTM

File: cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/template/etl/realtime/source/JmsSource.java [hunk 2]
```diff
@@ -111,24 +112,26 @@ public void initialize(RealtimeContext context) throws Exception {
     // Get environment vars - this would be prefixed with java.naming.*
     final Hashtable<String, String> envVars = new Hashtable<String, String>();
     for (Map.Entry<String, String> entry : runtimeArguments.entrySet()) {
-      if (entry.getKey() != null && entry.getKey().startsWith(JAVA_NAMING_PREFIX)) {
-        envVars.put(entry.getKey(), entry.getValue());
-      }
+      envVars.put(entry.getKey(), entry.getValue());
```
Comment (gsps1) [inline cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/template/etl/realtime/source/JmsSource.java]: key can't be null is it?

Comment (henry-cask) [inline cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/template/etl/realtime/source/JmsSource.java]: No, key can not be null from config properties

Comment (gsps1) [inline cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/template/etl/realtime/source/JmsSource.java]: ok

**Pull request merged**